Content.Server/Tesla/EntitySystem/TeslaEnergyBallSystem.cs

```diff
@@ -0,0 +1,86 @@
+using Content.Server.Administration.Logs;
+using Content.Server.Singularity.Components;
+using Content.Server.Tesla.Components;
+using Content.Shared.Database;
+using Content.Shared.Singularity.Components;
+using Content.Shared.Mind.Components;
+using Content.Shared.Tag;
+using Robust.Shared.Physics.Events;
+using Content.Server.Lightning.Components;
+using Robust.Server.Audio;
+
+namespace Content.Server.Tesla.EntitySystems;
+
+/// <summary>
+/// A component that tracks an entity's saturation level from absorbing other creatures by touch, and spawns new entities when the saturation limit is reached.
+/// </summary>
+public sealed class TeslaEnergyBallSystem : EntitySystem
+{
+    [Dependency] private readonly IAdminLogManager _adminLogger = default!;
+    [Dependency] private readonly AudioSystem _audio = default!;
+    [Dependency] private readonly TagSystem _tagSystem = default!;
+
+    public override void Initialize()
+    {
+        base.Initialize();
+
+        SubscribeLocalEvent<TeslaEnergyBallComponent, StartCollideEvent>(OnStartCollide);
+    }
+
+    public override void Update(float frameTime)
+    {
+        base.Update(frameTime);
+
+        var query = EntityQueryEnumerator<TeslaEnergyBallComponent>();
+        while (query.MoveNext(out var uid, out var teslaEnergyBall))
+        {
+            teslaEnergyBall.AccumulatedFrametime += frameTime;
+
+            if (teslaEnergyBall.AccumulatedFrametime < teslaEnergyBall.UpdateInterval)
+                continue;
+
+            AdjustEnergy(uid, teslaEnergyBall, -teslaEnergyBall.EnergyLoss * teslaEnergyBall.AccumulatedFrametime);
+            teslaEnergyBall.AccumulatedFrametime = 0f;
+        }
+    }
+
+    private void OnStartCollide(Entity<TeslaEnergyBallComponent> tesla, ref StartCollideEvent args)
+    {
+        if (TryComp<ContainmentFieldComponent>(args.OtherEntity, out var field))
+            return;
+        if (TryComp<LightningComponent>(args.OtherEntity, out var lightning)) //its dirty. but idk how to setup colliders to not eating lightnings
+            return;
```

Comment:
Do you even need TryComp here?

Reply:
if it's removed, the tesla will eat its own mini electra balls.